generator/abstractmetabuilder.cpp

```diff
@@ -402,7 +404,10 @@ bool AbstractMetaBuilder::build()
         return false;
 
     QTextStream stream(&file);
-    stream.setCodec(QTextCodec::codecForName("UTF-8"));
+#   if QT_VERSION < QT_VERSION_CHECK(6,0,0)
+        stream.setCodec(QTextCodec::codecForName("UTF-8"));
+        /* Note required in Qt6: see the same call in asttoxml.cpp */
+#   endif
     QByteArray contents = stream.readAll().toUtf8();
```

Comment:
Why is a QTextStream used here anyway, if the read content is converted back to UTF-8 immediately?
It seems to me the content of the file could be read without the intermediate decoding/encoding step.

Reply:
Maybe to remove the BOM which would probably mess up the resultant byte array?  It also handles UTF-16 correctly in Qt6.